mdata/store_cassandra.go

```diff
@@ -584,9 +588,14 @@ LOOP:
 		}
 		err := outcome.i.Close()
 		if err != nil {
+			if err == context.Canceled || err == context.DeadlineExceeded {
+				// query was aborted.
+				return nil, nil
+			}
 			tracing.Failure(span)
 			tracing.Error(span, err)
 			errmetrics.Inc(err)
+			return nil, err
```

Comment:
for better or worse, it used to be by design that we would continue here and just use as much data as we could get back.  remember the days before @replay started, and we would see charts with missing data here and there because certain chunks timed out? we would see the timeout errors on our MT dashboard, and MT always tried to make the most out of it.  
anyway the importance of "correct or error" (as opposed to "best effort") is become more and more clear as we see with some customers and their alerting queries. so :+1: